iredis/client.py

```diff
@@ -43,69 +86,44 @@ def __init__(self, host, port, db, password=None):
         self.host = host
         self.port = port
         self.db = db
-        if config.decode:
-            self.connection = Connection(
-                host=self.host,
-                port=self.port,
-                db=self.db,
-                password=password,
-                encoding=config.decode,
-                decode_responses=True,
-                encoding_errors="replace",
-                socket_keepalive=config.socket_keepalive,
-            )
-        else:
-            self.connection = Connection(
-                host=self.host,
-                port=self.port,
-                db=self.db,
-                password=password,
-                decode_responses=False,
-                socket_keepalive=config.socket_keepalive,
-            )
+        self._password = password
+        self._encoding = config.decode
+        self.manager = ConnectionManager(
+            host, port, db, password
+        )
+
         # all command upper case
         self.answer_callbacks = command2callback
         self.callbacks = self.reder_funcname_mapping()
-        try:
-            self.connection.connect()
-        except Exception as e:
-            print(str(e), file=sys.stderr)
-        if not config.no_info:
-            try:
-                self.get_server_info()
-            except Exception as e:
-                logger.warning(f"[After Connection] {str(e)}")
-                config.no_version_reason = str(e)
-        else:
-            config.no_version_reason = "--no-info flag activated"
 
-    def get_server_info(self):
-        self.connection.send_command("INFO")
+    async def get_server_version(self):
+        conn = await self.manager.get_connection()
+        result = await conn.execute("INFO")
         # safe to decode Redis's INFO response
-        info_resp = utils.ensure_str(self.connection.read_response())
+        info_resp = utils.ensure_str(result)
 
         version = re.findall(r"^redis_version:([\d\.]+)\r\n", info_resp, re.MULTILINE)[
             0
         ]
         logger.debug(f"[Redis Version] {version}")
-        config.version = version
+        return version
```

Comment:
Here if we don't set `config.version`, `HELP` for command since won't work.

https://github.com/laixintao/iredis/blob/60e58e12f6ec7a4b93febb4a449e9da6b2cab15a/iredis/client.py#L375